### ISSUE TITLE
match-declaration-order does not work

### ISSUE THREAD

**Issue by pmattke** [palantir/tslint#4020]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
interface IStuff {
    b: string; 
    a: string;
    c: string;
}

const stuff: IStuff = {
    a: "123",
    b: "12345",
    c: "foobar"
};
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "extends": ["tslint:recommended"],
    "jsRules": {},
    "rules": {
        "arrow-parens": false,
        "forin": false,
        "interface-name": {
            "options": ["always-prefix"]
        },
        "max-line-length": [true, 120],
        "no-shadowed-variable": false,
        "no-string-literal": false,
        "no-trailing-whitespace": false,
        "no-var-requires": false,
        "object-literal-sort-keys": [true, "match-declaration-order"],
        "only-arrow-functions": false,
        "ordered-imports": false,
        "space-before-function-paren": false,
        "trailing-comma": false,
        "variable-name": {
            "options": ["allow-leading-underscore"]
        }
    },
    "rulesDirectory": []
}

```

#### Actual behavior

The stuff object is not initialized with the property order used in IStuff. So, according to my config, I expect tslint to complain about that.

#### Expected behavior

TSLint does not complain at all. Even when I change the order in the literal, so it neither matches alphabetical order, nor interface order.

Am I getting something wrong or is this a bug?

**Comment by johnwiseheart** [comment]
The rule requires type information, and so is not supported in VSCode. If run with type info (`--project`) the code run is https://github.com/palantir/tslint/blob/29fbace7bd67195c7e7f1f115a3170e092f4e78c/src/rules/objectLiteralSortKeysRule.ts#L93 so the rule is actually used.

**Comment by johnwiseheart** [comment]
When you run `tslint` with Type Information, it means that it loads all of the files into memory so that it can compare the types between files. This is not possible by just linting individual files. See https://github.com/Microsoft/vscode-tslint/issues/70 for more information about why VSCode doesn't support it.
